File: Medium/Python/48. Rotate Image.py
class Solution(object):
    def rotate(self, matrix):
        for i in range((len(matrix) + 1) / 2):
            for j in range(len(matrix) / 2):
                temp = matrix[i][j]
                matrix[i][j] = matrix[len(matrix) - j - 1][i]
                matrix[len(matrix) - j - 1][i] = matrix[len(matrix) - i - 1][len(matrix) - j - 1]
                matrix[len(matrix) - i - 1][(len(matrix) - j - 1)] = matrix[j][len(matrix) - i - 1]
                matrix[j][len(matrix) - i - 1] = temp

File: Medium/Java/48. Rotate Image.java
class Solution {
    public void rotate(int[][] matrix) {
        for(int i = 0; i < (matrix.length + 1) / 2; i++){
            for(int j = 0; j < matrix.length / 2; j++){
                int temp = matrix[i][j];
                matrix[i][j] = matrix[matrix.length - j - 1][i];
                matrix[matrix.length - j - 1][i] = matrix[matrix.length - i - 1][matrix.length - j - 1];
                matrix[matrix.length - i - 1][matrix.length - j - 1] = matrix[j][matrix.length - i - 1];
                matrix[j][matrix.length - i - 1] = temp;
            }
        }
    }
}
